modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/provider/test/APIApplicationProviderTest.java

```diff
@@ -162,6 +165,46 @@ public void test() throws Exception {
 
 		Assert.assertEquals(
 			"name ne 'testName'", filter.getODataFilterString());
+
+		String baseURL = RandomTestUtil.randomString();
+
+		HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"applicationStatus", "published"
+			).put(
+				"baseURL", baseURL
+			).put(
+				"title", "test"
+			).toString(),
+			"headless-builder/applications/" +
+				apiApplicationJSONObject.getLong("id"),
+			Http.Method.PUT);
+
+		apiApplication = _apiApplicationProvider.fetchAPIApplication(
+			baseURL, TestPropsValues.getCompanyId());
+
+		Assert.assertEquals(baseURL, apiApplication.getBaseURL());
+		Assert.assertEquals("test", apiApplication.getTitle());
+
+		HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"applicationStatus", "unpublished"
+			).put(
+				"baseURL", "test"
+			).put(
+				"title", "title"
+			).toString(),
+			"headless-builder/applications/by-external-reference-code/" +
+				_API_APPLICATION_ERC,
+			Http.Method.PUT);
+
+		APIApplicationTestUtil.assertNotDeployedAPIApplication("test");
+
+		apiApplication = _apiApplicationProvider.fetchAPIApplication(
```

Comment:
I wonder if we should fetch the APIApplication here by baseURL and the companyId. We've just modified the baseURL so we want to make sure the APIApplication of the id that we wanted to modify is actually changed. Maybe we can use the getPublishedAPIApplications to figure out whether the app of a specific ID has the status and the baseURL that we want it to have? Does that make sense?

Reply:
> So, what we could to here is to check that the API application works with the updated baseURL and doesn't work with the old one.
> An assert of status code when calling the endpoint with this changed baseURL would be good proof of this.
I see different code on assert status code, one is assert jsonObject.status, the other is _assertSuccessfulHttpCode in HeadlessBuilderResourceTest.java, can we use  assertSuccessfulHttpCode instead of jsonObject.status?